server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -150,56 +149,52 @@ private void extensionsDiscovery() throws IOException {
-                                        extension.getName(),
-                                        extension.getUniqueId(),
-                                        // placeholder for ephemeral id, will change with POC discovery
-                                        extension.getUniqueId(),
-                                        extension.getHostName(),
-                                        extension.getHostAddress(),
-                                        new TransportAddress(TransportAddress.META_ADDRESS, Integer.parseInt(extension.getPort())),
-                                        new HashMap<String, String>(),
-                                        Version.fromString(extension.getVersion()),
-                                        pluginInfo
-                                    )
-                                );
-                                logger.debug("Loaded extension: " + extension);
-                                break;
-                            } catch (IllegalArgumentException e) {
-                                logger.error(e.toString());
-                            }
-                        }
-                    }
-
-                } catch (final IOException e) {
-                    throw new IOException("Could not load plugin descriptor " + plugin.getFileName(), e);
+                    extensionsList.add(
+                        new DiscoveryExtension(
+                            extension.getName(),
+                            extension.getUniqueId(),
+                            // placeholder for ephemeral id, will change with POC discovery
+                            extension.getUniqueId(),
+                            extension.getHostName(),
+                            extension.getHostAddress(),
+                            new TransportAddress(InetAddress.getByName(extension.getHostAddress()), Integer.parseInt(extension.getPort())),
+                            new HashMap<String, String>(),
+                            Version.fromString(extension.getOpensearchVersion()),
+                            new PluginInfo(
+                                extension.getName(),
+                                extension.getDescription(),
+                                extension.getVersion(),
+                                Version.fromString(extension.getOpensearchVersion()),
+                                extension.getJavaVersion(),
+                                extension.getClassName(),
+                                new ArrayList<String>(),
+                                Boolean.parseBoolean(extension.hasNativeController())
+                            )
+                        )
+                    );
+                    logger.info("Loaded extension: " + extension);
+                } catch (IllegalArgumentException e) {
+                    logger.error(e.toString());
                 }
             }
+            logger.info("Loaded all extensions");
```

Comment:
We should add a check here before logging the message. Case: If there's only one extension and even that gets caught in the catch block above.